pkg/aws/cloud.go

```diff
@@ -114,9 +108,41 @@ func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer) (Cloud,
 		metricsCollector.InjectHandlers(&sess.Handlers)
 	}
 
+	ec2Service := services.NewEC2(sess)
+
+	if len(cfg.VpcID) == 0 {
+		nodeName := os.Getenv("NODENAME")
```

Comment:
a nit comment, given this function is a bit long, 
maybe created a dedicated file like `pkg/aws/utils/vpc_id.go`, and have functions like
```
inferVPCID(sess, metadata), which invokes inferVPCIDFromNodeName
inferVPCIDFromNodeName(sess, nodeName)
```

Reply:
Didn't spot this comment, but ended up doing pretty much the same you describe here as it makes the fallback logic easier. I put the function below `NewCloud` though.